### PR TITLE
Respect the user's package format selection when creating new projects

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug
Fixes: https://github.com/NuGet/Home/issues/6736
Regression: No  
If Regression then when did it last work:   Never worked, missed in PR
If Regression then how are we preventing it in future:   

## Fix
Details: 
Respect the customer's selected default package management format during template creation if the given project template supports Package Reference

## Testing/Validation
Tests Added: Not yet.
Reason for not adding tests: Will add tests in this PR soon. Just wanted to get eyes on the logic.  
Validation done:  Manual, will add automated tests.

### CODE REVIEW

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs [hunk 1]
```diff
@@ -160,18 +161,24 @@ internal string GetRegistryRepositoryPath(string keyName, IEnumerable<IRegistryK
             IVsPackageInstaller packageInstaller,
             EnvDTE.Project project,
             PreinstalledPackageConfiguration configuration,
+            bool preferPackageReferenceFormat,
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs:165]: add xml docs for the variable.

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs [hunk 2]
```diff
@@ -160,18 +161,24 @@ internal string GetRegistryRepositoryPath(string keyName, IEnumerable<IRegistryK
...
             ThreadHelper.ThrowIfNotOnUIThread();
 
-            string repositoryPath = configuration.RepositoryPath;
+            var repositoryPath = configuration.RepositoryPath;
             var repositorySource = new Configuration.PackageSource(repositoryPath);
             var failedPackageErrors = new List<string>();
 
             // find the project
             var defaultProjectContext = new VSAPIProjectContext();
+
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs]: nit: remove extra line.

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs [hunk 3]
```diff
@@ -160,18 +161,24 @@ internal string GetRegistryRepositoryPath(string keyName, IEnumerable<IRegistryK
...
             var nuGetProject = await _solutionManager.GetOrCreateProjectAsync(project, defaultProjectContext);
+            if (preferPackageReferenceFormat && await NuGetProjectUpgradeUtility.IsNuGetProjectUpgradeableAsync(nuGetProject, project, false))
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs]: pass the bool as a named parameter.

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/VsTemplateWizard.cs [hunk 4]
```diff
@@ -266,9 +266,12 @@ private Task ProjectItemFinishedGeneratingAsync(ProjectItem projectItem)
             {
                 if (configuration.Packages.Any())
                 {
+                    var packageManagementFormat = new PackageManagementFormat(_settings);
+                    
                     await _preinstalledPackageInstaller.PerformPackageInstallAsync(_installer,
                         project,
                         configuration,
+                        packageManagementFormat.SelectedPackageManagementFormat == 1,
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/VsTemplateWizard.cs]: Move this to a bool and leave a comment stating that 1 means package reference.

**Review by mishra14 — APPROVED** [review]

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs (hunk 3 above)
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs]: What is need for this boolean? when would we install package through these api in existing project?

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs]: What do you mean by an existing project?
I'm making that call to not duplicate the logic of checking whether a project GUID is PR compatible.

**Comment by nkolev92** [comment]
🔔 @mishra14 @jainaashish

**Review by mishra14 — APPROVED** [review]

**Review by jainaashish — APPROVED** [review]

**Pull request merged**